File: java/src/main/java/io/fivexer/sdk/model/WorkerMetricsPeriod.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * One worker's rolling window: recorded time, throughput and response counters.
 *
 * <p>Medians are null for a worker with no history, and {@code acceptanceRate} is null before any
 * offer — a rate over nothing is not zero.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerMetricsPeriod {
    private String from;
    private String to;
    private List<WorkerMetricsWindowDay> days;
    private Long medianWaitMs;
    private Long medianCycleMs;
    private int shiftCount;
    private long onShiftMs;
    private long breakMs;
    private long workingMs;
    private int offered;
    private int accepted;
    private int rejected;
    private int completed;
    private int failed;
    private int expired;
    private int released;
    private Double acceptanceRate;

    public String getFrom() { return from; }
    public String getTo() { return to; }

    /** The per-day completion counts, or null when the window is not broken down. */
    public List<WorkerMetricsWindowDay> getDays() { return days; }

    public Long getMedianWaitMs() { return medianWaitMs; }
    public Long getMedianCycleMs() { return medianCycleMs; }
    public int getShiftCount() { return shiftCount; }

    /** Time on shift in the window, breaks included. */
    public long getOnShiftMs() { return onShiftMs; }

    public long getBreakMs() { return breakMs; }

    /** {@code onShiftMs - breakMs} — worked time. */
    public long getWorkingMs() { return workingMs; }

    public int getOffered() { return offered; }
    public int getAccepted() { return accepted; }
    public int getRejected() { return rejected; }
    public int getCompleted() { return completed; }
    public int getFailed() { return failed; }
    public int getExpired() { return expired; }
    public int getReleased() { return released; }

    /** accepted / offered over the window; null before any offer, never zero. */
    public Double getAcceptanceRate() { return acceptanceRate; }
}

File: php/src/Model/WorkerMetricsPeriod.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * One worker's rolling window: recorded time, throughput and response counters.
 *
 * Medians are null for a worker with no history, and `acceptanceRate` is null before any offer —
 * a rate over nothing is not zero.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerMetricsPeriod
{
    public function __construct(
        public readonly string $from,
        public readonly string $to,
        /** @var list<WorkerMetricsWindowDay>|null the per-day counts, or null when not broken down */
        public readonly ?array $days,
        public readonly ?int $medianWaitMs,
        public readonly ?int $medianCycleMs,
        public readonly int $shiftCount,
        /** Time on shift in the window, breaks included. */
        public readonly int $onShiftMs,
        public readonly int $breakMs,
        /** onShiftMs − breakMs, worked time. */
        public readonly int $workingMs,
        public readonly int $offered,
        public readonly int $accepted,
        public readonly int $rejected,
        public readonly int $completed,
        public readonly int $failed,
        public readonly int $expired,
        public readonly int $released,
        /** accepted / offered over the window; null before any offer, never zero. */
        public readonly ?float $acceptanceRate,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            from: (string) ($data['from'] ?? ''),
            to: (string) ($data['to'] ?? ''),
            days: Json::parseEachOrNull($data, 'days', [WorkerMetricsWindowDay::class, 'fromArray']),
            medianWaitMs: isset($data['medianWaitMs']) ? (int) $data['medianWaitMs'] : null,
            medianCycleMs: isset($data['medianCycleMs']) ? (int) $data['medianCycleMs'] : null,
            shiftCount: (int) ($data['shiftCount'] ?? 0),
            onShiftMs: (int) ($data['onShiftMs'] ?? 0),
            breakMs: (int) ($data['breakMs'] ?? 0),
            workingMs: (int) ($data['workingMs'] ?? 0),
            offered: (int) ($data['offered'] ?? 0),
            accepted: (int) ($data['accepted'] ?? 0),
            rejected: (int) ($data['rejected'] ?? 0),
            completed: (int) ($data['completed'] ?? 0),
            failed: (int) ($data['failed'] ?? 0),
            expired: (int) ($data['expired'] ?? 0),
            released: (int) ($data['released'] ?? 0),
            acceptanceRate: isset($data['acceptanceRate']) ? (float) $data['acceptanceRate'] : null,
        );
    }
}
